hook/scriptwrapper.go

```diff
@@ -126,21 +129,45 @@ func NewScriptWrapper(opts ...scriptWrapperOpt) (*ScriptWrapper, error) {
 		return nil, fmt.Errorf("Hook path was not provided")
 	}
 
-	var err error
-	var isBashHook bool
-	var isPwshHook bool
+	// Extract any shebang line from the hook to copy into the wrapper.
+	shebang, err := utils.ShebangLine(wrap.hookPath)
+	if err != nil {
+		return nil, fmt.Errorf("Hook path could not be read: %w", err)
+	}
+
+	// Previously we assumed Bash, because the wrapper relied on a Bash-ism.
+	// By using `bk-agent env dump`, the wrapper is compatible with /bin/sh.
+	//
+	// If there is no shebang line, the decision on what shell to use is the
+	// responsibility of the job executor.
+	//
+	// But if the shebang specifies something weird like Ruby 🤪
+	// the wrapper won't work. Stick to POSIX shells for now.
+	if shebang != "" {
+		// Splitting the shebang by spaces (ignoring quotes, etc) should be
+		// good enough to determine the POSIX-ness of the interpreter.
+		shebs := strings.Fields(strings.TrimPrefix(shebang, "#!"))
+		if !utils.IsPOSIXShell(shebs) {
+			return nil, fmt.Errorf("Hook starts with an unsupported shebang line %q", shebang)
+		}
```

Comment:
I think it's better to split the shebang line in the `utils.IsPOSIXShell` method. It seems like it should know more about what to do with the shebang line than this package.

Also, the two methods in `utils` seem quite related. It might be a good idea to make the package name more specific. A `shebang` package perhaps?

Reply:
Sounds good to me! Done.